storage/src/nodestore.rs

```diff
@@ -1711,8 +1721,12 @@ impl<S: ReadableStorage> Iterator for FreeListIterator<'_, S> {
         };
 
         // update the next address to the next free block
+        let parent_ptr = std::mem::replace(
```

Comment:
Take a look at PrevPeekable https://docs.rs/prev-iter/latest/prev_iter/struct.PrevPeekable.html

You might be able to use this or at least implement it the same way. I think parent_ptr is too heavy and names ending in `_ptr` usually do not add any value (what is wrong with parent?)